pkg/solana/chain.go

```diff
@@ -154,13 +152,26 @@ func (v *verifiedCachedClient) verifyChainID(ctx context.Context) (bool, error)
 		return v.chainIDVerified, fmt.Errorf("failed to fetch ChainID in verifiedCachedClient: %w", err)
 	}
 
-	// check chainID matches expected chainID
-	expectedChainID := strings.ToLower(v.expectedChainID)
 	// if this is localnet, allow any chain ID as long as it's not spoofing an official network
-	ignore := v.chainID == "localnet" && !slices.Contains([]string{"mainnet", "testnet", "devnet"}, expectedChainID)
-	if !ignore && v.chainID != expectedChainID {
-		v.chainIDVerified = false
-		return v.chainIDVerified, fmt.Errorf("client returned mismatched chain id (expected: %s, got: %s): %s", expectedChainID, v.chainID, v.nodeURL)
+	ignore := v.expectedChainID == "localnet"
+
+	if !ignore {
+		var matches bool
+		switch v.chainID {
+		case "devnet":
+			matches = client.DevnetGenesisHash == v.expectedChainID
```

Comment:
My proposal was that if it is really a problem to simultaneously expect all NOPs and toml configs to be updated with genesis hashes, then we can do this here temporarily:

case "devnet":
			matches = (client.DevnetGenesisHash == v.expectedChainID) || (v.chainID == v.expectedChainID)

.
.
.
similarly other cases.

This way, we handle both cases gracefully and give everyone time to upgrade.

Reply:
But if we can get everyone to update their configs then this change is good.

Reply:
Actually I think this is wrong, v.chainID is returned from client.ChainID() and it would be gensis hash or localnet.

We should do switch case for v.expectedChainID

Reply:
So in this case, default is handling the gensishash matching, and all other case are for backwards compatibility